### PR TITLE
Provision KVs for applications tier, store logon username/password in KV

### PR DESCRIPTION
**Pull request by YunziZhangAzure**
## Problem
<Please describe what problem this PR is trying to resolve>
#775

## Solution
<Please elaborate the solution for the problem>
1. Create two KVs for applications.

2. If authentication type = password, a password will be generated unless user specifies one.

## Tests
<Please provide steps to test the PR>

## Notes
<Additional comments for the PR>
Currently ansible_inventory in output_files module doesn't have password printed out. Will have another PR enable this feature to print the password to hosts.yml and hosts.

This pr also modifies the output of hdb module. The reason is because when trying to deploy only application tier, some of the output of hdb module is not empty list which causes the output_files module fail. Will do more tests about this point.

Retrieving the ssh key from saplandscape is not included in this pr.

### CODE REVIEW

File: deploy/terraform/terraform-units/modules/sap_system/app_tier/vm-app.tf [hunk 1]
```diff
@@ -48,9 +49,12 @@ resource "azurerm_linux_virtual_machine" "app" {
...
   }
 
-  admin_ssh_key {
-    username   = local.authentication.username
-    public_key = file(var.sshkey.path_to_public_key)
+  dynamic "admin_ssh_key" {
+    for_each = range(local.enable_auth_password ? 0 : 1)
+    content {
+      username   = local.sid_auth_username
+      public_key = file(var.sshkey.path_to_public_key)
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/app_tier/vm-app.tf:56]: Same question as hdb. This should use the keypair from sap_landscape.

**Review by rukawata — COMMENTED** [review]
Overall looks good! Same question as the other.

File: deploy/terraform/terraform-units/modules/sap_system/app_tier/vm-app.tf (hunk 1 above)
Comment (YunziZhangAzure) [reply deploy/terraform/terraform-units/modules/sap_system/app_tier/vm-app.tf:56]: Will be included in a future pr

**Review by rukawata — APPROVED** [review]
lgtm!

**Pull request merged**